lib/nodes/decorators/While.ts

```diff
@@ -44,14 +50,19 @@ export class While<S extends BlueshellState, E> extends Decorator<S, E> {
 			return res;
 		} else if (storage.break) {
 			// teardown internal state and yield to the behavior tree because the loop has completed
-			storage.break = undefined;
+			if (storage.lastLoopResult) {
+				// Parent will see one additiona event than the child when it evaluates the conditional
```

Comment:
One small typo here (additional).  If I understand this correctly, this makes it so when we break out of the loop and proceed, we still see the result of the child (but we wouldn't see the results of any of its children (because all children are cleared prior to going back up to the top to recheck the loop condition).  This seems like a good compromise.

Reply:
yeah, i just realize that this only fixes the state of the direct child, and not it's descendants. To fix that, we would need an api to set lastEventSeen on each "final" child at every descendent level. It's not that hard, but probably not worth it.

Reply:
actually, this isn't too hard. I'll take a crack at it